File: 2390_removing_stars_from_strings/solution.go
package main

func solution(s string) string {
    stack := make([]byte, 0, len(s))

    for i := 0; i < len(s); i++ {
        if s[i] != '*' {
            stack = append(stack, s[i])
        } else if len(stack) > 0 {
            stack = stack[:len(stack)-1]
        }
    }

    return string(stack)
}

File: 2390_removing_stars_from_strings/solution.mjs
import assert from "node:assert";
import { describe, it } from "node:test";

const testCases = [
  { input: "leet**cod*e", expected: "lecoe" },
  { input: "erase*****", expected: "" },
  { input: "abc*def*ghi*", expected: "abdegh" },
];

function solution(s) {
  const stack = [];

  for (const char of s) {
    if (char !== "*") {
      stack.push(char);
    } else if (stack.length) {
      stack.pop();
    }
  }

  return stack.join("");
}

describe("Remove stars from string", () => {
  testCases.forEach(({ input, expected }) => {
    it(`"${input}" should return "${expected}"`, () => {
      assert.strictEqual(solution(input), expected);
    });
  });
});
